### DISCUSSION TITLE
Raspberry Pi Zero: pipx install succeeds, invoking causes "Illegal Instruction"?

### DISCUSSION THREAD

**Discussion by rsend** [mitmproxy/mitmproxy#6163] [Q&A]
Hi,
Just tried installing mitmproxy on a Raspberry Pi Zero running Dietpi. I installed pipx (works, verified with other packages) and then installed mitmproxy (also seems to have worked). It tells me I can now use the three programs, but if I try invoking any of them, I get "Illegal Instruction" and nothing further.

pipx reports it installed v 9.0.1 using Python 3.9.1, if that helps.

Since DietPi by definition doesn't include much, are there any dependencies I should be aware of or something?

Cheers,
R

**Comment by mhils** [comment]
For running mitmproxy on a Raspberry Pi, my strong recommendation is to go with a 64bit OS and then use our Docker images (which have an aarch64 variant). With any other configuration you're likely in for some trouble.

**Comment by nneonneo** [comment]
If this is a RPi Zero (as opposed to RPi Zero 2 or any newer version), it's running ARM6, which is incompatible with software that assumes ARM7 or above.

I'm not sure where the incompatible binaries would be coming from, though, other than _maybe_ the `mitmproxy_rs` bits, but those don't come prebuilt for 32-bit ARM.
